Source/WebCore/style/StyleFontSizeFunctions.cpp

```diff
@@ -184,21 +184,32 @@ int legacyFontSizeForPixelSize(int pixelFontSize, bool shouldUseFixedDefaultSize
     return findNearestLegacyFontSize<float>(pixelFontSize, fontSizeFactors, mediumSize);
 }
 
-static float adjustedFontSize(float size, float sizeAdjust, float xHeight)
+static float adjustedFontSize(float size, float sizeAdjust, float metricValue)
 {
     if (!size)
         return 0;
 
-    float aspectValue = xHeight / size;
+    float aspectValue = metricValue / size;
     return size * (sizeAdjust / aspectValue);
 }
 
-float adjustedFontSize(float size, float sizeAdjust, const FontMetrics& metrics)
+float adjustedFontSize(float size, const FontSizeAdjust& sizeAdjust, const FontMetrics& metrics)
 {
-    if (!metrics.hasXHeight())
-        return size;
+    switch (sizeAdjust.metric) {
+    case FontSizeAdjust::Metric::CapHeight:
+        return metrics.hasCapHeight() ? adjustedFontSize(size, *sizeAdjust.value, metrics.floatCapHeight()) : size;
+    case FontSizeAdjust::Metric::ChWidth:
+        return metrics.zeroWidth() ? adjustedFontSize(size, *sizeAdjust.value, *metrics.zeroWidth()) : size;
+        break;
+    case FontSizeAdjust::Metric::IcWidth:
```

Comment:
This fallthrough doesn't seem right; we probably want to at least add a FIXME with a link to a bug about adding a real metric here. (Or to remove IcWidth from the spec)

Reply:
I added FIXME for further investigation here. When I tried, Gecko handled ic-height and ic-width in the same way, so I adopted the same approach.